src/efs_parser/BaseFile.py

```diff
@@ -47,16 +52,36 @@ def __init__(self, file):
         }
         self.byteRanges = {}
 
-    def is_local(self, file):
+    def split_s3_components(self,filename):
+        phase1 = filename.replace("s3://", "")
+        region_start = phase1.find("@")
+        if region_start == -1:
+            raise Exception('Invalid S3 file name - missing region name')
+        file_name_start = phase1.find("/")
+        if file_name_start == -1:
+            raise Exception('Invalid S3 file name - missing file name')
+        bucket_name = phase1[:file_name_start]
+        file_name = phase1[file_name_start + 1:region_start]
+        region = phase1[region_start + 1:]
+        return bucket_name, region, file_name
+
+    def get_file_source(self, file):
         """
         Checks if file is local or hosted publicly
 
         Args:
             file: location of file
         """
         if "http://" in file or "https://" in file or "ftp://" in file:
-            return False
-        return True
+            return "http"
+        elif "s3://" in file:
+            bucket_name, region, file_name = self.split_s3_components(file)
```

Comment:
split should be called in the constructor

Reply:
Done